scenarioColors.js

```diff
@@ -1,42 +1,62 @@
 const ScenarioSdk = require('@qasymphony/scenario-sdk');
 
-const StepSdk = {
-    getStepSdk(qtestToken, scenarioProjectId) {
-        return new ScenarioSdk.Steps({ qtestToken, scenarioProjectId });
-    }
+// @pure
+const state = {
+    sdk: null
 }
 
-const Steps = {
-    updateStepResults(stepSdk, name, status) {
-        return stepSdk.getSteps(`"${name}"`).
-            then(steps => Promise.all(steps.map(step => stepSdk.updateStep(step.id, Object.assign(step, { status })))))
-            .catch(function (err) {
-                console.log('Error updating colors: ' + err);
-            });
-    }
-};
-
-exports.handler = function ({ event: body, constants, triggers }, context, callback) {
-    var payload = body;
-    var testLogs = payload.logs;
-
-    stepSdk = StepSdk.getStepSdk(constants.QTEST_TOKEN, constants.SCENARIO_PROJECT_ID);
-    
-    for (var res of testLogs) {
-        for (var step of res["test_step_logs"]) {
-            var stepName = step.description;
-            var stepStatus = step.status;
-
-            // Undefined means no step definition existed and it should fail
-            if (stepStatus == "undefined") {
-                stepStatus = "failed";
-            }
-
-            // one of PASSED (green), FAILED (red), or SKIPPED (yellow)
-            stepStatus = stepStatus.toUpperCase();
-
-            // Call the pulse API to update step results
-            Steps.updateStepResults(stepSdk, stepName, stepStatus);
-        }
-    }
+// @impure: Data returned is dependent on where the ScenarioSDK is pointing
+const getStepSdk = (qtestToken, scenarioProjectId) => new ScenarioSdk.Steps({ qtestToken, scenarioProjectId });
+
+// @impure: has side effects (updates ScenarioSdk)
+const setConfigAndRetrieveSDK = (scenarioUrl, qtestToken, scenarioProjectId) => {
+    ScenarioSdk.config({ scenarioUrl: scenarioUrl });
+    return getStepSdk(qtestToken, scenarioProjectId);
 }
+
+// @impure: uses I/O 
+const loggingOutError = err => console.log('Error updating colors: ' + err);
+
+// @impure: reliant on state object
+const addingStatusToStep = (sdkStep, status) => state.sdk.updateStep(sdkStep.id, Object.assign(sdkStep, { "status": status }));
```

Comment:
Please put this on multiple lines, my eyes, they hurts :p

Reply:
Done